packages/mui-material-next/src/IconButton/IconButton.tsx

```diff
@@ -3,43 +3,52 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { chainPropTypes, unstable_capitalize as capitalize } from '@mui/utils';
-import { unstable_composeClasses as composeClasses } from '@mui/base';
-import { alpha } from '@mui/system';
-import ButtonBase from '@mui/material/ButtonBase';
-import styled from '../styles/styled';
-import useThemeProps from '../styles/useThemeProps';
-import iconButtonClasses, { getIconButtonUtilityClass } from './iconButtonClasses';
+import { unstable_composeClasses as composeClasses, useSlotProps } from '@mui/base';
+import { useThemeProps, alpha, shouldForwardProp } from '@mui/system';
+import resolveProps from '@mui/utils/resolveProps';
+import ButtonBase from '../ButtonBase';
+import { styled } from '../styles';
+import { getIconButtonUtilityClass } from './iconButtonClasses';
+import buttonBaseClasses from '../ButtonBase/buttonBaseClasses';
+import { IconButtonOwnerState, IconButtonProps, IconButtonTypeMap } from './IconButton.types';
+import { ButtonGroupButtonContext, ButtonGroupContext } from '../ButtonGroup';
 
-const useUtilityClasses = (ownerState) => {
-  const { classes, disabled, color, edge, size } = ownerState;
+const useUtilityClasses = (ownerState: IconButtonOwnerState) => {
+  const { classes, color, edge, size } = ownerState;
 
   const slots = {
     root: [
       'root',
-      disabled && 'disabled',
-      color !== 'default' && `color${capitalize(color)}`,
+      `color${capitalize(color ?? '')}`,
```

Comment:
`color` also has a default value of `'default'` so `?? ''` isn't needed

I think we need to keep the `color !== 'default'` condition since `colorDefault` isn't an expected className: https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/IconButton/iconButtonClasses.ts#L13-L26